sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/VersioningTest.groovy

```diff
@@ -0,0 +1,364 @@
+        blobClient.getVersionClient(blobItemV2.getVersionId()).getAppendBlobClient().download(outputV2)
+
+        then:
+        outputV1.toByteArray() == contentV1.getBytes(StandardCharsets.UTF_8)
+        outputV2.toByteArray() == contentV2.getBytes(StandardCharsets.UTF_8)
+    }
+
+    def "Delete Block Blob by Version"() {
+        given:
+        def inputV1 = new ByteArrayInputStream(contentV1.getBytes(StandardCharsets.UTF_8))
+        def inputV2 = new ByteArrayInputStream(contentV2.getBytes(StandardCharsets.UTF_8))
+        def blobItemV1 = blobClient.getBlockBlobClient().upload(inputV1, inputV1.available())
+        def blobItemV2 = blobClient.getBlockBlobClient().upload(inputV2, inputV2.available(), true)
+
+        when:
+        blobClient.getVersionClient(blobItemV1.getVersionId()).getBlockBlobClient().delete()
+
+        then:
+        !blobClient.getVersionClient(blobItemV1.getVersionId()).exists()
+        blobClient.getVersionClient(blobItemV2.getVersionId()).exists()
+    }
+
+    def "Delete Page Blob by Version"() {
+        given:
+        def contentV1 = getRandomByteArray(PageBlobClient.PAGE_BYTES)
+        def contentV2 = getRandomByteArray(PageBlobClient.PAGE_BYTES)
+        def inputV1 = new ByteArrayInputStream(contentV1)
+        def inputV2 = new ByteArrayInputStream(contentV2)
+        def blobItemV1 = blobClient.getPageBlobClient().create(PageBlobClient.PAGE_BYTES)
+        blobClient.getPageBlobClient().uploadPages(new PageRange().setStart(0).setEnd(PageBlobClient.PAGE_BYTES-1), inputV1)
+        def blobItemV2 = blobClient.getPageBlobClient().create(PageBlobClient.PAGE_BYTES, true)
+        blobClient.getPageBlobClient().uploadPages(new PageRange().setStart(0).setEnd(PageBlobClient.PAGE_BYTES-1), inputV2)
+
+        when:
+        blobClient.getVersionClient(blobItemV1.getVersionId()).getPageBlobClient().delete()
+
+        then:
+        !blobClient.getVersionClient(blobItemV1.getVersionId()).exists()
+        blobClient.getVersionClient(blobItemV2.getVersionId()).exists()
+    }
+
+    def "Delete Append Blob by Version"() {
+        given:
+        def inputV1 = new ByteArrayInputStream(contentV1.getBytes(StandardCharsets.UTF_8))
+        def inputV2 = new ByteArrayInputStream(contentV2.getBytes(StandardCharsets.UTF_8))
+        def blobItemV1 = blobClient.getAppendBlobClient().create()
+        blobClient.getAppendBlobClient().appendBlock(inputV1, inputV1.available())
+        def blobItemV2 = blobClient.getAppendBlobClient().create(true)
+        blobClient.getAppendBlobClient().appendBlock(inputV2, inputV2.available())
+
+        when:
+        blobClient.getVersionClient(blobItemV1.getVersionId()).getAppendBlobClient().delete()
+
+        then:
+        !blobClient.getVersionClient(blobItemV1.getVersionId()).exists()
+        blobClient.getVersionClient(blobItemV2.getVersionId()).exists()
+    }
+
+    def "Retrieve Block Blob Properties by Version"() {
```

Comment:
nit: we usually use download instead of retrieve and get properties instead of retrieve properties for test naming

Reply:
Ok.